genesyscloud/routing_email_route/genesyscloud_routing_email_route_proxy.go

```diff
@@ -92,73 +92,65 @@ func (p *routingEmailRouteProxy) deleteRoutingEmailRoute(ctx context.Context, do
+func getAllRoutingEmailRouteByDomainIdFn(_ context.Context, p *routingEmailRouteProxy, domains []platformclientv2.Inbounddomain, pattern string) (*map[string][]platformclientv2.Inboundroute, *platformclientv2.APIResponse, error) {
+	const pageSize = 100
 	var allInboundRoutes = make(map[string][]platformclientv2.Inboundroute)
 	var apiResponse *platformclientv2.APIResponse
-	for _, domain := range *domains.Entities {
-		var allDomainRoutes []platformclientv2.Inboundroute
+	for _, domain := range domains {
+
+		var allDomainRoutes = make([]platformclientv2.Inboundroute, 0)
 
-		for pageNum := 1; pageNum <= *domains.PageCount; pageNum++ {
-			routes, resp, err := p.routingApi.GetRoutingEmailDomainRoutes(*domain.Id, 100, pageNum, name)
+		for pageNum := 1; ; pageNum++ {
+			routes, resp, err := p.routingApi.GetRoutingEmailDomainRoutes(*domain.Id, pageSize, pageNum, pattern)
 			if err != nil {
 				apiResponse = resp
-				return nil, apiResponse, fmt.Errorf("Failed to get routing email route: %s", err)
+				return nil, apiResponse, fmt.Errorf("failed to get routing email route: %s", err.Error())
 			}
 			if routes.Entities == nil || len(*routes.Entities) == 0 {
 				break
 			}
 			allDomainRoutes = append(allDomainRoutes, *routes.Entities...)
-			allInboundRoutes[*domain.Id] = allDomainRoutes
 		}
+		allInboundRoutes[*domain.Id] = allDomainRoutes
 	}
 	return &allInboundRoutes, apiResponse, nil
 }
 
 // getAllRoutingEmailRouteFn is the implementation for retrieving all routing email route in Genesys Cloud
-func getAllRoutingEmailRouteFn(ctx context.Context, p *routingEmailRouteProxy, domainId string, name string) (*map[string][]platformclientv2.Inboundroute, *platformclientv2.APIResponse, error) {
-	var allInboundRoutes = make(map[string][]platformclientv2.Inboundroute)
+func getAllRoutingEmailRouteFn(ctx context.Context, p *routingEmailRouteProxy, domainId, pattern string) (*map[string][]platformclientv2.Inboundroute, *platformclientv2.APIResponse, error) {
 	const pageSize = 100
 	var apiResponse *platformclientv2.APIResponse
 
-	domains, resp, err := p.routingApi.GetRoutingEmailDomains(pageSize, 1, false, domainId)
-	if err != nil {
-		return nil, resp, fmt.Errorf("Failed to get routing email domains: %s", err)
+	var allDomains = make([]platformclientv2.Inbounddomain, 0)
+	for pageNum := 1; ; pageNum++ {
+		domains, resp, err := p.routingApi.GetRoutingEmailDomains(pageSize, pageNum, false, domainId)
+		apiResponse = resp
+		if err != nil {
+			return nil, resp, fmt.Errorf("failed to get routing email domains: %s", err.Error())
+		}
+		if domains.Entities == nil || len(*domains.Entities) == 0 {
+			break
+		}
+		allDomains = append(allDomains, *domains.Entities...)
 	}
-	if domains.Entities == nil || len(*domains.Entities) == 0 {
-		return &allInboundRoutes, resp, nil
+
+	if len(allDomains) == 0 {
+		return nil, apiResponse, nil
```

Comment:
Because we now return `nil` here, that means any calling functions need to also check that this is not nil before processing the records. The `readRoutingEmailRoute()` and `getAllRoutingEmailRoutes()` functions both make calls to this function without checking that the inboundRoutes result is null.

Thus we get https://inindca.atlassian.net/browse/DEVTOOLING-1099